microprofile/opentracing-smallrye/src/main/java/org/wildfly/extension/microprofile/opentracing/smallrye/TracingDeploymentProcessor.java

```diff
@@ -0,0 +1,124 @@
+package org.wildfly.extension.microprofile.opentracing.smallrye;
+
+import io.smallrye.opentracing.SmallRyeTracingDynamicFeature;
+import org.jboss.as.ee.weld.WeldDeploymentMarker;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.server.deployment.Phase;
+import org.jboss.as.web.common.WarMetaData;
+import org.jboss.as.weld.deployment.WeldPortableExtensions;
+import org.jboss.metadata.javaee.spec.ParamValueMetaData;
+import org.jboss.metadata.web.jboss.JBossWebMetaData;
+import org.jboss.metadata.web.spec.ListenerMetaData;
+import org.wildfly.extension.microprofile.opentracing.smallrye.deployment.TracerInitializer;
+import org.wildfly.extension.microprofile.opentracing.smallrye.deployment.TracingCDIExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TracingDeploymentProcessor implements DeploymentUnitProcessor {
+  public static final Phase PHASE = Phase.POST_MODULE;
+  public static final int PRIORITY = 0x4000;
+
+  @Override
+  public void deploy(DeploymentPhaseContext deploymentPhaseContext) {
+    TracingLogger.ROOT_LOGGER.processingDeployment();
+    DeploymentUnit deploymentUnit = deploymentPhaseContext.getDeploymentUnit();
+
+    if (!WeldDeploymentMarker.isPartOfWeldDeployment(deploymentUnit)) {
+      // SmallRye JAX-RS requires CDI. Without CDI, there's no integration needed
+      TracingLogger.ROOT_LOGGER.noCdiDeployment();
```

Comment:
Should this be logging an error or throwing an exception? If it requires CDI it seems the deployment should fail.

Reply:
Good point. As tracing is usually not something that is a requirement for a business code to run (like logging), we tend to just warn users instead of failing their applications.

@objectiser , what do you think? Should we just warn, or should we fail the deployment?

Reply:
My preference would be just to warn at this stage.